test/test_rosbag/CMakeLists.txt

```diff
@@ -50,6 +50,8 @@ if(CATKIN_ENABLE_TESTING)
   add_rostest(${PROJECT_BINARY_DIR}/test/latched_sub.test)
   add_rostest(test/record_two_publishers.test)
   add_rostest(test/record_one_publisher_two_topics.test)
+  add_rostest(test/record_sigint_cleanup.test)
+  add_rostest(test/record_sigterm_cleanup.test)
```

Comment:
Beside the name of the test both `.test` files contain the same content? Are the same tests (embedded in the `record_cleanup.py` file) run twice?

Reply:
You are correct, this is my mistake. See the latest refactor that addresses this. Here's my (truncated) output when running tests locally:

```
      Start  9: _ctest_test_rosbag_rostest_test_record_sigint_cleanup.test

9: Test command: /home/ubuntu/ws/build/test_rosbag/catkin_generated/env_cached.sh "/usr/bin/python2" "/home/ubuntu/ws/install/catkin/share/catkin/cmake/test/run_tests.py" "/home/ubuntu/ws/build/test_rosbag/test_results/test_rosbag/rostest-test_record_sigint_cleanup.xml" "--return-code" "/usr/bin/python2 /home/ubuntu/ws/install/rostest/share/rostest/cmake/../../../bin/rostest --pkgdir=/home/ubuntu/ws/src/test_rosbag --package=test_rosbag --results-filename test_record_sigint_cleanup.xml --results-base-dir "/home/ubuntu/ws/build/test_rosbag/test_results" /home/ubuntu/ws/src/test_rosbag/test/record_sigint_cleanup.test "
9: Test timeout computed to be: 9.99988e+06
9: ... logging to /home/ubuntu/.ros/log/rostest-ip-172-31-31-214-2489.log
9: [ROSUNIT] Outputting test results to /home/ubuntu/ws/build/test_rosbag/test_results/test_rosbag/rostest-test_record_sigint_cleanup.xml
9: [Testcase: testtest_sigint_cleanup] ... ok
9: 
9: [ROSTEST]-----------------------------------------------------------------------
9: 
9: [test_rosbag.rosunit-test_sigint_cleanup/test_sigint_cleanup][passed]
9: 
9: SUMMARY
9:  * RESULT: SUCCESS
9:  * TESTS: 1
9:  * ERRORS: 0
9:  * FAILURES: 0
9: 
9: rostest log file is in /home/ubuntu/.ros/log/rostest-ip-172-31-31-214-2489.log
9: -- run_tests.py: execute commands
9:   /usr/bin/python2 /home/ubuntu/ws/install/rostest/share/rostest/cmake/../../../bin/rostest --pkgdir=/home/ubuntu/ws/src/test_rosbag --package=test_rosbag --results-filename test_record_sigint_cleanup.xml --results-base-dir "/home/ubuntu/ws/build/test_rosbag/test_results" /home/ubuntu/ws/src/test_rosbag/test/record_sigint_cleanup.test 
9: -- run_tests.py: verify result "/home/ubuntu/ws/build/test_rosbag/test_results/test_rosbag/rostest-test_record_sigint_cleanup.xml"
 9/14 Test  #9: _ctest_test_rosbag_rostest_test_record_sigint_cleanup.test .............   Passed   11.53 sec
test 10
      Start 10: _ctest_test_rosbag_rostest_test_record_sigterm_cleanup.test

10: Test command: /home/ubuntu/ws/build/test_rosbag/catkin_generated/env_cached.sh "/usr/bin/python2" "/home/ubuntu/ws/install/catkin/share/catkin/cmake/test/run_tests.py" "/home/ubuntu/ws/build/test_rosbag/test_results/test_rosbag/rostest-test_record_sigterm_cleanup.xml" "--return-code" "/usr/bin/python2 /home/ubuntu/ws/install/rostest/share/rostest/cmake/../../../bin/rostest --pkgdir=/home/ubuntu/ws/src/test_rosbag --package=test_rosbag --results-filename test_record_sigterm_cleanup.xml --results-base-dir "/home/ubuntu/ws/build/test_rosbag/test_results" /home/ubuntu/ws/src/test_rosbag/test/record_sigterm_cleanup.test "
10: Test timeout computed to be: 9.99988e+06
10: ... logging to /home/ubuntu/.ros/log/rostest-ip-172-31-31-214-2584.log
10: [ROSUNIT] Outputting test results to /home/ubuntu/ws/build/test_rosbag/test_results/test_rosbag/rostest-test_record_sigterm_cleanup.xml
10: [Testcase: testtest_sigterm_cleanup] ... ok
10: 
10: [ROSTEST]-----------------------------------------------------------------------
10: 
10: [test_rosbag.rosunit-test_sigterm_cleanup/test_sigterm_cleanup][passed]
10: 
10: SUMMARY
10:  * RESULT: SUCCESS
10:  * TESTS: 1
10:  * ERRORS: 0
10:  * FAILURES: 0
10: 
10: rostest log file is in /home/ubuntu/.ros/log/rostest-ip-172-31-31-214-2584.log
10: -- run_tests.py: execute commands
10:   /usr/bin/python2 /home/ubuntu/ws/install/rostest/share/rostest/cmake/../../../bin/rostest --pkgdir=/home/ubuntu/ws/src/test_rosbag --package=test_rosbag --results-filename test_record_sigterm_cleanup.xml --results-base-dir "/home/ubuntu/ws/build/test_rosbag/test_results" /home/ubuntu/ws/src/test_rosbag/test/record_sigterm_cleanup.test 
10: -- run_tests.py: verify result "/home/ubuntu/ws/build/test_rosbag/test_results/test_rosbag/rostest-test_record_sigterm_cleanup.xml"
10/14 Test #10: _ctest_test_rosbag_rostest_test_record_sigterm_cleanup.test ............   Passed   11.53 sec
test 11

```